### ISSUE TITLE
Snowflake connection expires in connection pool

### ISSUE THREAD

**Issue by virgilp** [snowflakedb/snowflake-connector-net#527]
### Issue description
We have a long-running process that infrequently creates a new Snowflake connection and tries to run queries.
The problem that we have with it is that after a while, the connection we get is a connection from the Snowflake connection pool - but that connection has the token expired. When executing first query, the Snowflake connector tries to renew the session and fails:

### Example code
```cs
N/A
```

### Error log
```
{"date":"2022-09-23T07:03:23.2808361+00:00","level":"INFO","logger":"Snowflake.Data.Core.SFSession","message":"Renew the session."}
{"date":"2022-09-23T07:03:23.2953586+00:00","level":"ERROR","logger":"Snowflake.Data.Core.SFSession","message":"Renew session failed","exception":"Snowflake.Data.Client.SnowflakeDbException (0x80004005): Error: Authentication token has expired. The user must authenticate again. SqlState: , VendorCode: 390114, QueryId: "}
{"date":"2022-09-23T07:03:23.2981152+00:00","level":"ERROR","logger":"Snowflake.Data.Core.SFStatement","message":"Query execution failed.","exception":"Snowflake.Data.Client.SnowflakeDbException (0x80004005): Error: Authentication token has expired. The user must authenticate again. SqlState: , VendorCode: 390114, QueryId: \n at Snowflake.Data.Core.SFSession.renewSession()\n at Snowflake.Data.Core.SFStatement.Execute(Int32 timeout, String sql, Dictionary`2 bindings, Boolean describeOnly)"}
```

### Configuration
*Driver version:*
2.0.15
*Dotnet framework and version:*
.NET 6.0.4

*Server version:* 6.31.1

*Client OS:* Linux 5.4.0-1089-azure #94~18.04.1-Ubuntu SMP  x86_64 GNU/Linux

**Comment by virgilp** [comment]
I see this is actually well known issue.

**Comment by ChrisIsidora** [comment]
@sfc-gh-igarish When will this fix be released to 2.0.17?

**Comment by sfc-gh-igarish** [comment]
The fix has been merged 3 days ago, so fix will available in October release.
